### PR TITLE
[release/6.0] Run tests for internal azdo PRs

### PR DESCRIPTION
**Pull request by safern**
This enables running test for internal PRs. Fixes https://github.com/dotnet/runtime/issues/47328

Infra only change. 

Backport of https://github.com/dotnet/runtime/pull/65603
Test official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1624984&view=results

### CODE REVIEW

File: eng/pipelines/common/variables.yml [hunk 1]
```diff
@@ -10,9 +10,9 @@ variables:
   value: 1000
 
 - name: isOfficialBuild
-  value: ${{ and(ne(variables['System.TeamProject'], 'public'), ne(variables['Build.Reason'], 'PullRequest')) }}
+  value: ${{ and(eq(variables['System.TeamProject'], 'internal'), eq(variables['Build.DefinitionName'], 'dotnet-runtime-official')) }}
```
Comment (hoyosjs) [inline eng/pipelines/common/variables.yml:13]: What def are we using internal PRs too? Probably this should also have !pr checks .

**Review by hoyosjs — COMMENTED** [review]
The only other question is how are mono+clr passing a creator and token?

File: eng/pipelines/common/variables.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/common/variables.yml:13]: We are using one called `runtime` but I think it is a good idea to be defensive and consider PR reason.

**Comment by safern** [comment]
Mono and clr already had that setup. They use this template: https://github.com/dotnet/runtime/blob/d1f4d238089c5f5a6ea36e1bcd828e5d15557f3f/eng/pipelines/common/templates/runtimes/send-to-helix-step.yml#L44

i.e: https://github.com/dotnet/runtime/blob/3fc8b099443a8b84bd158b8e49e38be6dd941180/eng/pipelines/common/templates/runtimes/run-test-job.yml#L378

**Review by hoyosjs — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
@ericstj please merge this as well, the failures are unrelated

**Pull request merged**